nixos/modules/services/network-filesystems/ipfs.nix

```
@@ -280,7 +280,7 @@ in
         User = cfg.user;
         Group = cfg.group;
         StateDirectory = "";
-        ReadWritePaths = [ "" cfg.dataDir ];
+        ReadWritePaths = optionals cfg.autoMount [ "" cfg.dataDir ];
```

```suggestion
        ReadWritePaths = optionals (!cfg.autoMount) [ "" cfg.dataDir ];
```
This condition needs to be inverted.